unittest/UMEUnitTestCommon.h

```diff
@@ -8649,8 +8650,9 @@ void genericRSHVTest_random()
 
         for (int i = 0; i < VEC_LEN; i++) {
             inputB[i] = randomValue<SCALAR_UINT_TYPE>(gen) % MAX_BIT_COUNT;
-            SCALAR_UINT_TYPE inputA_range = 1 << MAX_BIT_COUNT;
-            inputA[i] = randomValue<SCALAR_TYPE>(gen) % inputA_range;
+            uint64_t inputA_range = 1 << MAX_BIT_COUNT;
+            inputA[i] = randomValue<SCALAR_TYPE>(gen);
+	    inputA[i] = inputA_range == 0 ? inputA[i] : inputA[i] % inputA_range;

```

Comment:
inputA_range should never be 0, as long as '1 << MAX_BIT_COUNT' is interpreted by the compiler as  uint64_t. In this case I think that compiler interprets it as 'int32_t' expression, therefore for a value of MAX_BIT_COUNT higher than 32, the top bits are simply discarded.

I think that a proper fix would be to replace the declaration of inputA_range with:

uint64_t inputA_range = uint64_t(1) << uint64_t(MAX_BIT_COUNT);

this way there should be no ambiguity in the way the compiler selects the resultant type.

The same comment applies for other shift tests.